### PR TITLE
Add Windows 10 UWP implementation

### PR DESCRIPTION
**Pull request by robinmanuelthiel**
**Min Supported Version:** 
Windows 10.0 (Build 10240)

**Known issues:**
- ~~Testing Mode currently not available for MvvmCross (needs a platform wide change)~~
- Does not support subscriptions (as UWP does not support this, yet)

### CODE REVIEW

**Comment by jamesmontemagno** [comment]
Hey there! I have been adjusting the API recently so if you can update based on this new finalized API, should be better. I will start a review on your current and give you some feedback.

File: src/Plugin.InAppBilling.UWP/InAppBillingImplementation.cs [hunk 1]
```diff
@@ -10,57 +13,128 @@ namespace Plugin.InAppBilling
     /// </summary>
     public class InAppBillingImplementation : IInAppBilling
     {
+        private bool testingMode;
+
+        /// <param name="testingMode">UWP offers a way to test in-app purchases with the CurrentAppSimulator class instead of CurrentApp.</param>
+        public InAppBillingImplementation(bool testingMode = false)
```
Comment (robinmanuelthiel) [inline src/Plugin.InAppBilling.UWP/InAppBillingImplementation.cs]: Yep, adding `IsTesting` sounds way better. Will do that.

File: src/Plugin.InAppBilling.UWP/InAppBillingImplementation.cs [hunk 2]
```diff
@@ -10,57 +13,128 @@ namespace Plugin.InAppBilling
...
         /// </summary>
         /// <returns>Task to disconnect</returns>
-        public Task DisconnectAsync()
+        public Task DisconnectAsync() => Task.CompletedTask;
+
+        public async Task<IEnumerable<InAppBillingProduct>> GetProductInfoAsync(ItemType itemType, params string[] productIds)
         {
-            throw new NotImplementedException();
+            // Get list of products from store or simulator
+            var listingInformation = testingMode ? await CurrentAppSimulator.LoadListingInformationAsync() : await CurrentApp.LoadListingInformationAsync();
```
Comment (robinmanuelthiel) [inline src/Plugin.InAppBilling.UWP/InAppBillingImplementation.cs]: Great idea, will do that.

**Comment by robinmanuelthiel** [comment]
Hey @jamesmontemagno, thanks for your review. Sounds good, I will update the PR during the day and give you an update, when it's ready to get reviewed again!

File: src/Plugin.InAppBilling.UWP/InAppBillingImplementation.cs [hunk 3]
```diff
@@ -1,7 +1,10 @@
 using Plugin.InAppBilling.Abstractions;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
+using System.Xml;
+using Windows.ApplicationModel.Store;
 
 namespace Plugin.InAppBilling
```
Comment (robinmanuelthiel) [inline src/Plugin.InAppBilling.UWP/InAppBillingImplementation.cs]: I don't really get, what you mean here. UWP needs them both to work with IAPs. How could we remove them? Build works for me here. Any advice?

**Comment by robinmanuelthiel** [comment]
Alright, I have updated the PR. Beside failing AppVeyor build. We have to take a look at that together, I guess...

File: src/Plugin.InAppBilling.UWP/InAppBillingImplementation.cs (hunk 3 above)
Comment (jamesmontemagno) [inline src/Plugin.InAppBilling.UWP/InAppBillingImplementation.cs]: If you look in the project... essentially the same file is linked to older Windows Phone projects. this means the build will fail. You need to create a blank implementation for those projects.

**Comment by jamesmontemagno** [comment]
I created a new UWP branch to look into it and will help out with me trying to get it working. Should be good now :)

**Pull request merged**